### ISSUE TITLE
Work around destructuring types

### ISSUE THREAD

**Issue by evmar** [angular/tsickle#263]
This code:
```ts
function get({ bar = 0 }) { }
get({});
```

may produce an error like:
```
WARNING - Property bar never defined on $jscomp$destructuring$var1
```

if the name `bar` is never used elsewhere.
We can work around this either by providing a type annotation that mentions `bar` or perhaps by some throwaway code like
```
namespace __unused { var bar; }
```

**Comment by evmar** [comment]
This output has no warning:
```js
/**
 * @param {{bar: (number|undefined)}} __0
 */
function get({ bar = 0 } = {}) { }
get({});
```

**Comment by mprobst** [comment]
`__0` is the parameter index, I presume?

**Comment by evmar** [comment]
I've already forgotten how this works but my recollection is that currently there's no way to name the destructuring parameter, so tsickle just invents the names as "__0", "__1" etc.  I'm not sure how Closure eventually matches them up, though I imagine it probably just matches the indexes in the @params array with the function args.  (Another way of saying this is that I think Closure ignores the variable names in the jsdoc for destructuring params.)

**Comment by concavelenz** [comment]
Thats right, it is ignored by the compiler, the name is only used to complete the JSDoc "form" and to name the parameter for any generated documentation.

**Comment by mprobst** [comment]
@evmar is this done?

**Comment by evmar** [comment]
No.

Given the input:
```js
  declare function destructures({a: number});
```
We were generating:
```js
/** ...closure stuff */
function destructures({a: number});
```
which causes a Closure crash, but now we generate
```js
/** ...closure stuff */
function destructures(__0);
```
which is still wrong (need to put "a" in the jsdoc) but at least avoids the crash.
